libraries/AP_Logger/LoggerMessageWriter.cpp

```diff
@@ -233,14 +233,28 @@ void LoggerMessageWriter_WriteSysInfo::process() {
         stage = Stage::RC_PROTOCOL;
         FALLTHROUGH;
 
-    case Stage::RC_PROTOCOL:
+    case Stage::RC_PROTOCOL: {
         const char *prot = hal.rcin->protocol();
         if (prot == nullptr) {
             prot = "None";
         }
         if (! _logger_backend->Write_MessageF("RC Protocol: %s", prot)) {
             return; // call me again
         }
+        stage = Stage::BATT_CYCLES;
+        FALLTHROUGH;
+    }
+    case Stage::BATT_CYCLES:
+    
+        const AP_BattMonitor &battery = AP::battery();
+        for (uint8_t i = 0; i < AP_BATT_MONITOR_MAX_INSTANCES; i++) {
+            uint16_t cycles = 0;
```

Comment:
```suggestion
            uint16_t cycles;
```

Entirely reasonable to assume that if `get_cycle_count` returns true then it has filled in its return parameter.